flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1356,6 +1361,61 @@ IntrinsicLibrary::genPresent(mlir::Type,
                                           fir::getBase(args[0]));
 }
 
+// SCAN 
+fir::ExtendedValue
+IntrinsicLibrary::genScan(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() >= 2);
+
+  // required string argument
```

Comment:
SCAN looks like INDEX in that you can use variant calls without descriptors. It would be good to add that at some point.